subsys/net/ip/route.c

```diff
@@ -713,9 +759,12 @@ net_route_mcast_lookup(struct in6_addr *group)
 		struct net_route_entry_mcast *route = &route_mcast_entries[i];
 
 		if (!route->is_used) {
-			if (net_ipv6_addr_cmp(group, &route->group)) {
-				return route;
-			}
+			continue;
+		}
+		if (net_ipv6_is_prefix(group->s6_addr,
+				route->group.s6_addr,
```

Comment:
Indendation seems bit off, can you verify that the parameters are aligned properly